float32 position
float32 speed
float32 effort
